src/groupByKey.ts

```diff
@@ -2,6 +2,20 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+type Item = object;
+
+export function groupByKey(items: Item[], key: keyof Item): GroupsMap<Item> {
+  const group: GroupsMap<Item> = {};
+
+  items.forEach((item) => {
+    const groupKey = item[key];
```

Comment:
```suggestion
    const groupKey = String(item[key]);
```

So you don't need your `type Item`